dev/main.go

```diff
@@ -73,30 +82,62 @@ func (dev *DaggerDev) CLI() *CLI {
 	return &CLI{Dagger: dev}
 }
 
+// Return the Dagger source code
+func (dev *DaggerDev) Source() *dagger.Directory {
+	if !dev.ModCodegen {
+		return dev.Src
+	}
+	// FIXME: build this list dynamically, by scanning the source for modules
+	modules := []string{
+		"dev",
+		"dev/dirdiff",
+		"dev/go",
+		"dev/golangci",
+		"dev/graphql",
+		"dev/markdown",
+		"dev/shellcheck",
+		"sdk/elixir/runtime",
+		// FIXME: broken
+		// "sdk/python/dev",
+		"sdk/python/runtime",
+		"sdk/typescript/dev",
+		"sdk/typescript/dev/node",
+		"sdk/typescript/runtime",
+	}
+	eg, _ := errgroup.WithContext(context.TODO())
+	layers := make(chan *dagger.Directory, len(modules))
```

Comment:
This code doesn't actually need to be parallelized. All of the operations in this errorgroup are actually already lazy.

Sorry, I missed this earlier.

Reply:
They may be lazy but I think there's enough overhead to slow me down. I added this after seeing a stair-shaped trace that seemed to support that.